### PR TITLE
dry part 1: GitHubGlue support for keys or url as parm

### PR DESCRIPTION
**Pull request by webstech**
This is part of a series to make GitGitGadget more generic for use with other repos.  As part of this it also starts to address issue #150.  

The `pullRequestKey` may not be the best name or object but it is also not the first name or iteration.  Suggestions are welcome.  The existing `parsePullRequestURL` should be able to go away with the next iteration.

A next step would be to update the `azure` function to add the repo name to the trigger parms.  This can then be handled in the pipeline to set a third parm of the repo.  A full key would then be available for PR processing.  

### test: ci-helper check last parm for value

The number of parms to GitHubGlue::updatePR could change.  This
removes a dependency on the number of call parameters.

### dependency: remove repo url from notes

Part of a set of changes to make GitGitGadget more supportive of    other projects.

### regex reduction: introduce PR keys object

First phase in removing reparsing of pull request URL.  This positions   `GitHubGlue` to accept a URL string or object for most methods.

### CODE REVIEW

File: lib/ci-helper.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { MailArchiveGitHelper } from "./mail-archive-helper";
 import { MailCommitMapping } from "./mail-commit-mapping";
 import { IMailMetadata } from "./mail-metadata";
 import { IPatchSeriesMetadata } from "./patch-series-metadata";
+import { getPullRequesKeyFromtURL } from "./pullRequestKey";
```
Comment (dscho) [inline lib/ci-helper.ts:14]: Typo? I think the `t` before `URL` belongs before the `Key` instead.

File: lib/ci-helper.ts [hunk 2]
```diff
@@ -834,11 +845,10 @@ export class CIHelper {
         return await mailArchiveGit.processMails(prFilter);
     }
 
-    private async getPRInfo(prNumber: number, pullRequestURL: string):
+    private async getPRInfo(_prNumber: number, pullRequestURL: string):
```
Comment (dscho) [inline lib/ci-helper.ts:848]: Why not remove the `prNumber` parameter altogether?

File: lib/github-glue.ts [hunk 3]
```diff
@@ -227,20 +226,20 @@ export class GitHubGlue {
...
 
         const status = await this.client.rest.pulls.createReplyForReviewComment(
             {
                 body: comment,
                 comment_id: id,
-                owner,
-                pull_number: nr,
-                repo,
+                owner: prKey.owner,
+                pull_number: prKey.number,
```
Comment (dscho) [inline lib/github-glue.ts:241]: If we called the attribute `prKey.pull_number`, we could use the `...prKey` syntax to DRY up the code even further (not only here, but elsewhere, too). Even if we don't, `pull_number: prKey.number, ...prKey` might work.

**Review by dscho — DISMISSED** [review]
I love it!

I left a few suggestions, hopefully you find them useful.

File: lib/ci-helper.ts (hunk 1 above)
Comment (webstech) [reply lib/ci-helper.ts:14]: Sorry, I usually catch that sort of thing.

File: lib/ci-helper.ts (hunk 2 above)
Comment (webstech) [reply lib/ci-helper.ts]: I was trying to minimize the changes but yes, that makes more sense.

File: lib/github-glue.ts (hunk 3 above)
Comment (webstech) [reply lib/github-glue.ts]: Great idea!  Felt awkward using a type as a name anyway.

I did not use `...prKey` when `issue_number` was part of the request.  Concerned about future rejection of unexpected parameters (pull_number).  Should this be a concern?

Comment (dscho) [reply lib/github-glue.ts]: > I did not use ...prKey when issue_number was part of the request.

Makes sense!

**Review by dscho — APPROVED** [review]
What a delightful PR. Thank you!

**Pull request merged**